server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -642,3 +640,34 @@ export function configureWebSocketServices(
         });
     });
 }
+
+function addAlfredTrace(message: IDocumentMessage, numberOfMessagesPerTrace: number,
+    clientId: string, tenantId: string, documentId: string) {
+    if (message && core.DefaultServiceConfiguration.enableTraces && sampleMessages(message, numberOfMessagesPerTrace)) {
+        if (message.traces === undefined) {
+            message.traces = [];
+        }
+        message.traces.push(
+        {
+            action: "start",
+            service: "alfred",
+            timestamp: Date.now(),
+        });
+
+        const lumberjackProperties = {
+            [BaseTelemetryProperties.tenantId]: tenantId,
+            [BaseTelemetryProperties.documentId]: documentId,
+            clientId,
+            clientSequenceNumber: message.clientSequenceNumber,
+            traces: message.traces,
+            opType: message.type,
+        };
+        Lumberjack.info(`Message received by Alfred.`, lumberjackProperties);
```

Comment:
Can this logs affect performance somehow? If the traces are being added to the operation here, being possible to retrieve them later, would it be a disadvantage to not have that log here?